### PR TITLE
VR 3

### PR DESCRIPTION
**Pull request by JoshuaMitchell**

### CODE REVIEW

File: .env-travis [hunk 1]
```diff
@@ -3,3 +3,4 @@ TESTING_DATABASE_URL=postgres://foo:bar@localhost/ksvotes_test
 SECRET_KEY=sekritsekritsekritsekrit
 APP_CONFIG=testing
 CRYPT_KEY=tGMAP3iPMyCaHO21WPGpYl4cFkUB7JHF8fBnvce6OzA=
+USPS_USER_ID=033GPSIM1278
```
Comment (karpet) [inline .env-travis]: is this a legit code? will we have a different one for prod? (i.e. keeping prod keys out of git)

File: app/services/steps/step_vr_3.py [hunk 2]
```diff
@@ -1,10 +1,72 @@
...
+
+
+
+	def validate_addresses(self):
+		"""
+		Return values of usps lookup.  Keep track of order of addresses if multiple provided.
+		"""
+		#always expect the current address
+		current_address = dict([
+			('address', self.form_payload.get('addr', '')),
```
Comment (karpet) [inline app/services/steps/step_vr_3.py]: is there an extra indent space on this block? I know python cares about whitespace but not sure if it needs the extra space for multiline statement.

File: app/services/steps/step_vr_3.py [hunk 3]
```diff
@@ -1,10 +1,72 @@
...
+			 ('zip_code', self.form_payload.get('zip')),
+			 ('address_extended', self.form_payload.get('unit'))
+		])
+		extra_addresses = []
+
+		#construct additional addresses for request and update address_order
+		if self.form_payload.get('has_prev_addr'):
+			extra_addresses.append(dict([
+			 ('address', self.form_payload.get('prev_addr', '')),
+ 		     ('city', self.form_payload.get('prev_city')),
```
Comment (karpet) [inline app/services/steps/step_vr_3.py]: the indents here seem irregular.

File: app/services/steps/step_vr_3.py [hunk 4]
```diff
@@ -1,10 +1,72 @@
...
+	 	     ('state', self.form_payload.get('prev_state')),
+	 		 ('zip_code', self.form_payload.get('prev_zip')),
+	 	     ('address_extended', self.form_payload.get('prev_unit'))
+			]))
+			self.address_order.append('prev_addr')
+
+		if self.form_payload.get('has_mail_addr'):
+			extra_addresses.append(dict([
+			 ('address', self.form_payload.get('mail_addr', '')),
+ 		     ('city', self.form_payload.get('mail_city')),
```
Comment (karpet) [inline app/services/steps/step_vr_3.py]: same indent thing here

File: app/services/steps/tests/test_step_vr_3.py [hunk 5]
```diff
@@ -1,37 +1,100 @@
 # from app.models import *
 # from app.services.steps import Step_VR_3
 #
-# def test_step_vr3_is_complete_false(app, session, client):
```
Comment (karpet) [inline app/services/steps/tests/test_step_vr_3.py]: are these tests all commented out because the USPS service isn't live?

**Comment by JoshuaMitchell** [comment]
@karpet Sorry this is not at all ready for review.  I was just merging in all the other steps so I can get this one caught up to master.

**Pull request merged**